OpenRA.Mods.RA/Traits/Disguise.cs

```diff
@@ -120,12 +120,12 @@ public string VoicePhraseForOrder(Actor self, Order order)
 			return order.OrderString == "Disguise" ? info.Voice : null;
 		}
 
-		public Color RadarColorOverride(Actor self)
+		public Color RadarColorOverride(Actor self, Color color)
 		{
 			if (!Disguised || self.Owner.IsAlliedWith(self.World.RenderPlayer))
-				return self.Owner.Color.RGB;
+				return color;
 
-			return AsPlayer.Color.RGB;
+			return color = Game.Settings.Game.UsePlayerStanceColors ? AsPlayer.PlayerStanceColor(self) : AsPlayer.Color.RGB;
```

Comment:
This doesn't work, because you dropped the whole `apparentOwner` stuff when you moved most of `SelectionBarsRenderable.GetHealthColor` to `Player.PlayerStanceColor`.